File: atcoder/abc168C_colon.cpp
// unihernandez22 & glum & Vicfred
// https://atcoder.jp/contests/abc168/tasks/abc168_c
// math
#include <ios>
#include <iostream>
#include <cmath>
#include <math.h>
#include <iomanip>

using namespace std;

int main() {
    cout << fixed << showpoint;
    cout << setprecision(23);

    long double a, b, h, m;

    cin >> a >> b >> h >> m;

    long double x = M_PI/6;
    long double y = M_PI/30;
    long double z = x/60;
    long double C = abs(x*h + z*m - y*m);
    /* x = factor para las horas, pues 5*pi/30 es una hora
     * y = factor para los minutos, 2*pi/60 = pi/30
     * */

    cout << sqrt(a*a+b*b-2.0L*a*b*cos(C)) << endl;

    return 0;
}

File: atcoder/abc168C_colon.py
# uninhm
# https://atcoder.jp/contests/abc168/tasks/abc168_c
# math
 
from math import pi, cos
 
a, b, h, m = map(int, input().split())
x = pi/6
y = pi/30
z = x/60
C = abs(x*h+z*m - y*m)
print(f"{(a**2 + b**2 - 2 * a*b * cos(C))**2**-1:.20f}")
